api/v1alpha1/httproutefilter_types.go

```diff
@@ -91,6 +106,26 @@ type HTTPPathModifier struct {
 	ReplaceRegexMatch *ReplaceRegexMatch `json:"replaceRegexMatch,omitempty"`
 }
 
+// +kubebuilder:validation:XValidation:message="setFromHeader must be nil if the type is not SetFromHeader",rule="!(has(self.setFromHeader) && self.type != 'SetFromHeader')"
+// +kubebuilder:validation:XValidation:message="setFromHeader must be specified for SetFromHeader type",rule="!(!has(self.setFromHeader) && self.type == 'SetFromHeader')"
+// +kubebuilder:validation:XValidation:message="setFromBackend must be nil if the type is not SetFromBackend",rule="!(has(self.setFromBackend) && self.type != 'SetFromBackend')"
+// +kubebuilder:validation:XValidation:message="setFromBackend must be specified for SetFromBackend type",rule="!(!has(self.setFromBackend) && self.type == 'SetFromBackend')"
+type HTTPHostnameModifier struct {
+	// +kubebuilder:validation:Enum=SetFromHeader;SetFromBackend
+	// +kubebuilder:validation:Required
+	Type HTTPHostnameModifierType `json:"type"`
+
+	// SetFromHeader indicates that the Host header value would be replaced with the value of the header specified in setFromHeader.
+	// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routeaction-host-rewrite-header
+	// +optional
+	SetFromHeader *string `json:"setFromHeader,omitempty"`
+
+	// SetFromBackend indicates that the Host header value would be replaced by the DNS name of the backend if it exists.
+	// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routeaction-auto-host-rewrite
+	// +optional
+	SetFromBackend *bool `json:"setFromBackend,omitempty"`
```

Comment:
What will happen when type=SetFromBackend && SetFromBacken=false ?

Reply:
right, it's redundant, i'll remove it =]